### PR TITLE
feat(taiko-client): catch raiko deserialization errors

### LINKED ISSUE #18638: feat(taiko-client): catch raiko deserialization errors

### PR DESCRIPTION
**Pull request by YoGhurt111**

### CODE REVIEW

**Review by davidtaikocha — APPROVED** [review]

File: packages/taiko-client/prover/proof_producer/zkvm_producer.go [hunk 1]
```diff
@@ -479,14 +479,18 @@ func (s *ZKvmProofProducer) requestBatchProof(
...
+
 	if output.Data.Status == ErrProofInProgress.Error() {
 		return nil, ErrProofInProgress
 	}
 	if output.Data.Status == StatusRegistered {
 		return nil, ErrRetry
 	}
 
-	if len(output.Data.Proof.Proof) == 0 {
+	if output.Data.Proof == nil || len(output.Data.Proof.Proof) == 0 {
```
Comment (mask-pp) [inline packages/taiko-client/prover/proof_producer/zkvm_producer.go:493]: Set `Data         *RaikoProofDataV2 `json:"data,omitempty"`` then just check `output.Data.Proof == nil` is enough.

File: packages/taiko-client/prover/proof_producer/sgx_producer.go [hunk 2]
```diff
@@ -308,14 +308,18 @@ func (s *SGXProofProducer) requestBatchProof(
...
+	}
 	if output.Data.Status == ErrProofInProgress.Error() {
 		return nil, ErrProofInProgress
 	}
 	if output.Data.Status == StatusRegistered {
 		return nil, ErrRetry
 	}
 
-	if len(output.Data.Proof.Proof) == 0 {
+	if output.Data.Proof == nil ||
```
Comment (mask-pp) [inline packages/taiko-client/prover/proof_producer/sgx_producer.go:321]: ditto

**Review by mask-pp — APPROVED** [review]
LGTM, just two optional comments

File: packages/taiko-client/prover/proof_producer/zkvm_producer.go (hunk 1 above)
Comment (YoGhurt111) [reply packages/taiko-client/prover/proof_producer/zkvm_producer.go:493]: Got it, but I still want to use this code.

**Pull request merged**